File: app/src/main/java/com/capstone/project_niyakneyak/data/alarm_resource/AlarmDao.java
package com.capstone.project_niyakneyak.data.alarm_resource;

import com.capstone.project_niyakneyak.data.alarm_model.Alarm;

import androidx.lifecycle.LiveData;
import androidx.room.*;

import java.util.List;

@Dao
public interface AlarmDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Alarm alarm);


    @Query("SELECT * FROM alarm_table ORDER BY alarmCode ASC")
    LiveData<List<Alarm>> getAlarms();

    @Update
    void update(Alarm alarm);

    @Query("DELETE FROM alarm_table")
    void deleteAll();

    @Query("DELETE FROM alarm_table WHERE alarmCode = :alarmCode")
    void delete(int alarmCode);
}

File: app/src/main/java/com/capstone/project_niyakneyak/data/alarm_resource/AlarmDao.kt
package com.capstone.project_niyakneyak.data.alarm_resource

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import com.capstone.project_niyakneyak.data.alarm_model.Alarm

@Dao
interface AlarmDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(alarm: Alarm?)

    @get:Query("SELECT * FROM alarm_table")
    val alarms: LiveData<List<Alarm>>

    @Update
    fun update(alarm: Alarm?)

    @Query("DELETE FROM alarm_table WHERE alarmCode = :alarmCode")
    fun delete(alarmCode: Int)

    @Query("DELETE FROM alarm_table")
    fun deleteAll()
}